src/txn/txn.c

```diff
@@ -440,11 +440,37 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 	WT_RET(__wt_config_gets_def(session, cfg, "read_timestamp", 0, &cval));
 	if (cval.len > 0) {
 #ifdef HAVE_TIMESTAMPS
+		bool round_to_oldest;
+		int cmp;
 		wt_timestamp_t ts;
+		WT_TXN_GLOBAL *txn_global;
 
+		txn_global = &S2C(session)->txn_global;
 		WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
-		WT_RET(__wt_timestamp_validate(session,
```

Comment:
With this change we should call __wt_timestamp_validate something as __wt_commit_timestamp_validate. That function was used to validate read_timestamp, but even if validation failed it would have given output saying commit_timestamp older than oldest .. . It was a result of trying to generalise that function for all timestamp validation but I guess we were not very successful at that. That function really only validates commit_timestamp.